crates/wasi/src/view.rs

```diff
@@ -0,0 +1,86 @@
+use crate::ctx::WasiCtx;
+use wasmtime::component::ResourceTable;
+
+pub trait IoView: Send {
+    /// Yields mutable access to the internal resource management that this
+    /// context contains.
+    ///
+    /// Embedders can add custom resources to this table as well to give
+    /// resources to wasm as well.
+    fn table(&mut self) -> &mut ResourceTable;
+}
+
+pub trait WasiView: IoView {
```

Comment:
Could this retain some of the documentation from before as well? Ideally with an example or links to other examples throughout the crate.